### PR TITLE
set timeouts for connect, socket and query

### PR DESCRIPTION
**Pull request by malaffoon**
Thoughts when doing this:
* connect and validation timeouts should be short, i chose 10s. Could be lower, what do you think?
* query timeout depends on app, either 30s or 120s
* socket timeout needs to be longer than query timeout (or else query won't be aborted), i added 10s to query timeout

### CODE REVIEW

File: exam-processor/src/main/resources/application.yml [hunk 1]
```diff
@@ -29,16 +29,20 @@ spring:
...
     password:
     testWhileIdle: true
     validationQuery: SELECT 1
+    validationQueryTimeout: 10000
     driverClassName: com.mysql.jdbc.Driver
     initialize: false
     maxActive: 12
     initialSize: 6
     maxWait: 10000
+    jdbcInterceptors: "QueryTimeoutInterceptor(queryTimeout=30)"
```
Comment (malaffoon) [inline exam-processor/src/main/resources/application.yml]: This is just how you do query timeouts with the apache tomcat connection pool.

File: group-processor/src/main/resources/application.yml [hunk 2]
```diff
@@ -42,15 +42,19 @@ spring:
...
     username: root
     password:
     testWhileIdle: true
     validationQuery: SELECT 1
+    validationQueryTimeout: 10000
     driverClassName: com.mysql.jdbc.Driver
     initialize: false
     maxActive: 10
     initialSize: 4
+    jdbcInterceptors: "QueryTimeoutInterceptor(queryTimeout=120)"
```
Comment (malaffoon) [inline group-processor/src/main/resources/application.yml]: Made this one 120s for really large group files.

File: exam-processor/src/main/resources/application.yml (hunk 1 above)
Comment (agorina) [reply exam-processor/src/main/resources/application.yml]: Nice!

File: migrate-olap/src/main/resources/application.yml [hunk 3]
```diff
@@ -64,11 +64,14 @@ spring:
       ?useSSL=${spring.migrate_datasource.use-ssl:false}\
       &useLegacyDatetimeCode=${spring.migrate_datasource.use-legacy-datetime-code:false}\
       &characterEncoding=${spring.migrate_datasource.character-encoding:utf8}\
+      &connectTimeout=${spring.migrate_datasource.connect-timeout:10000}\
+      &socketTimeout=${spring.migrate_datasource.socket-timeout:0}\
```
Comment (malaffoon) [inline migrate-olap/src/main/resources/application.yml]: Set migrate-olap to be unlimited, but have placeholders so it is easier to override if desired.

**Pull request merged**